### PR TITLE
weird error with require('something.json') and hyperxify 

### PR DESCRIPTION
**Pull request by epiloque**
To reproduce:

```js
'use strict'

const something = require('something.json')
const hyperx = require('hyperx')
const vdom = require('virtual-dom')
const h = hyperx(vdom.h)

console.log(h`<h1>clicked times</h1>`)

```

The error is

```js

.../node_modules/acorn/dist/acorn.js:1747
  throw err;
  ^

SyntaxError: Unexpected token (2:16)
    at Parser.pp.raise (.../node_modules/acorn/dist/acorn.js:1745:13)
    at Parser.pp.unexpected (.../node_modules/acorn/dist/acorn.js:2264:8)
    at Parser.pp.semicolon (.../node_modules/acorn/dist/acorn.js:2243:59)
    at Parser.pp.parseExpressionStatement (.../node_modules/acorn/dist/acorn.js:2677:8)
    at Parser.pp.parseStatement (.../node_modules/acorn/dist/acorn.js:2462:160)
    at Parser.pp.parseBlock (.../node_modules/acorn/dist/acorn.js:2692:21)
    at Parser.pp.parseStatement (.../node_modules/acorn/dist/acorn.js:2443:19)
    at Parser.pp.parseTopLevel (.../node_modules/acorn/dist/acorn.js:2379:21)
    at parse (.../node_modules/acorn/dist/acorn.js:101:12)
    at module.exports (.../node_modules/falafel/index.js:22:15)
    at DestroyableTransform.end [as _flush] (.../node_modules/hyperxify/index.js:56:15)
    at DestroyableTransform.<anonymous> (.../node_modules/hyperxify/node_modules/readable-stream/lib/_stream_transform.js:123:12)
    at DestroyableTransform.g (events.js:273:16)
    at emitNone (events.js:80:13)
    at DestroyableTransform.emit (events.js:179:7)
    at prefinish (.../node_modules/hyperxify/node_modules/readable-stream/lib/_stream_writable.js:501:12)

```

Thank you!

### CODE REVIEW

**Comment by yoshuawuyts** [comment]
ACK, can reproduce

**Comment by yoshuawuyts** [comment]
appears this isn't a problem with `hyperxify`, but with `acorn`. Any sort of `.json` require seems to go haywire

**Comment by yoshuawuyts** [comment]
Could be related to https://github.com/ternjs/acorn/issues/385

**Comment by epiloque** [comment]
babelify, envify, uglifyify are working properly, hyperxify breaks the build

**Comment by epiloque** [comment]
literalify solves the same [issue](https://github.com/pluma/literalify/issues/3) by [excluding json files](https://github.com/pluma/literalify/blob/master/index.js#L3) 

```js

var transformTools = require('browserify-transform-tools');

module.exports = transformTools.makeRequireTransform('literalify', {excludeExtensions: ['json']}, function(args, opts, cb) {
  if (opts.config && args[0] in opts.config) {
    return cb(null, opts.config[args[0]]);
  } else {
    return cb();
  }
});

```

**Comment by yoshuawuyts** [comment]
oh yeah, this could then be solved by matching `path.extname(file) === '.json'` and then just returning chunks as-is without running `falafel`. I don't think using `browserify-transform-tools` would help here as we're returning a stream directly. Does that make sense? Would you like to write a PR for that?

**Comment by epiloque** [comment]
was going that direction, let me try (:

**Comment by epiloque** [comment]
@yoshuawuyts would you verify that?

File: index.js [hunk 1]
```diff
@@ -52,8 +53,12 @@ module.exports = function (file, opts) {
...
   }
   function end () {
-    var src = Buffer.concat(bufs).toString('utf8')
-    this.push(falafel(src, { ecmaVersion: 6 }, walk).toString())
+	var src = Buffer.concat(bufs).toString('utf8')
+	if (path.extname(file) === '.json') {
+		this.push(src)
+	} else {
+		this.push(falafel(src, { ecmaVersion: 6 }, walk).toString())
+	}
```
Comment (yoshuawuyts) [inline index.js]: could you fix the indentation here? You're missing a space

**Comment by epiloque** [comment]
done (:

**Comment by yoshuawuyts** [comment]
ACK tests pass + patch LGTM :sparkles:

cc/ @JamesKyburz @substack could you merge this patch?

**Comment by epiloque** [comment]
Thanks @yoshuawuyts

**Comment by JamesKyburz** [comment]
@epiloque Good catch!

What do you think about returning a stream straightaway for json files in the same way that [brfs] does it?

* https://github.com/substack/brfs/blob/master/index.js#L9

[brfs]:https://github.com/substack/brfs

**Comment by epiloque** [comment]
@JamesKyburz much cleaner, thank you (:

**Comment by epiloque** [comment]
@JamesKyburz @substack could you merge this ?

**Comment by JamesKyburz** [comment]
@epiloque Thanks for the patch. merged.

**Comment by yoshuawuyts** [comment]
:sparkles::sparkles::sparkles::sparkles::sparkles:

**Pull request merged**